### ISSUE TITLE
Issue with form attachment being sent via email

### ISSUE THREAD

**Issue by rhukster** [getgrav/grav-plugin-form#333]
Discovered this while debugging another issue.  My scenario.  

1. I have a form with a file field
2. In `process:` there is an email being sent with `attachments: file_field` to attach the file.  

```
...
        fields:
            file_field:
                type: file
                label: 'Upload Image'
                destination: '@self'
                random_name: false
                avoid_overwriting: true
                accept:
                    - 'image/*'

        process:
            email:
                from: '{{ config.plugins.email.from }}'
                to: ['{{ config.plugins.email.from }}', '{{ form.value.email }}']
                subject: '[Artist] {{ form.value.name|e }}'
                body: '{% include ''forms/data.html.twig'' %}'
                attachments:
                  - file_field
...
```

In Grav 1.5 this worked fine.  In Grav 1.6 the file has not been uploaded yet when this email process is run, so it fails trying to attach the file. 

It seems the file should be uploaded before any processing is run so it's available.

**Comment by mahagr** [comment]
I had to change uploads to happen later in order to make the logic to work properly with object controlled saves, such as flex object (but also others). There should be fallback to force early file save in regular cases, but it looks like I missed email.

This case raises a question, though. The file can be uploaded only once, after that it will not be available anymore as it's being moved. How should we handle emails in those cases?

**Comment by mahagr** [comment]
Should be fixed in email plugin.
